pkg/controller/dnszone/azureactuator.go

```diff
@@ -201,5 +202,30 @@ func (a *AzureActuator) UpdateMetadata() error {
 
 // SetConditionsForError sets conditions on the dnszone given a specific error. Returns true if conditions changed.
 func (a *AzureActuator) SetConditionsForError(err error) bool {
-	return false // Not implemented for Azure yet.
+	// other conditions not implemented for Azure yet, so set generic condition
+	var cloudErrorsConds []hivev1.DNSZoneCondition
+	var cloudErrorsCondsChanged bool
+	if err == nil {
+		cloudErrorsConds, cloudErrorsCondsChanged = controllerutils.SetDNSZoneConditionWithChangeCheck(
+			a.dnsZone.Status.Conditions,
+			hivev1.GenericDNSErrorsCondition,
+			corev1.ConditionFalse,
+			dnsNoCloudErrorReason,
+			"No cloud errors occurred",
+			controllerutils.UpdateConditionIfReasonOrMessageChange,
```

Comment:
I noticed this was `controllerutils.UpdateConditionNever` for awsactuator [above](https://github.com/openshift/hive/pull/1432/files#diff-2257da48fedf6f2c5593b99c2ea746f86c885d246da2758eb98bb4b2e0038983R617). I'm still trying to wrap my head around how that whole thing works, but IIUC it won't make a difference since
- a status change will always trigger an update; and
- if the status hasn't changed, neither has the reason/message, since it's hardcoded on success.

Reply:
That's correct - it wouldn't matter in this instance whether it is `UpdateConditionIfReasonOrMessageChange` or `UpdateConditionNever`, since the next update would only be when the status is changed, but given that this is a generic error condition, I just wanted to keep the opportunity in the future for the condition status to still be True but for some other reason. Either way it wouldn't make a difference right now.